README_EN.md

```diff
@@ -0,0 +1,240 @@
+
+#### Register
+
+Via code:
+
+```Swift
+  let controllerManager = FRDControllerManager.sharedInstance
+  controllerManager.register(URL(string: "/frodo/firstview")!, clazz: FirstViewController.self)
+```
+
+Via plist file:
+
+```Swift
+  let plistPath = Bundle.main.path(forResource: "FRDIntentRegisters", ofType: "plist")
+  let controllerManager = FRDControllerManager.sharedInstance
+  controllerManager.register(plistFile: plistPath)
+```
+
+#### Launch view controller by class
+
+```Swift
+  let intent = FRDIntent(clazz: SecondViewController.self)
+  let manager = FRDControllerManager.sharedInstance
+  manager.startController(source: self, intent: intent)
+```
+
+#### Launch view controller by url
+
+```Swift
+  let intent = FRDIntent(uri: URL(string: "/frodo/firstview")!)
+  let manager = FRDControllerManager.sharedInstance
+  manager.startController(source: self, intent: intent)
+```
+
+#### Launch view controller to get result data
+
+```Swift
+  let intent = FRDIntent(clazz: ThirdViewController.self)
+  intent.putExtra(name: "text", data: "Text From Source")
+  let manager = FRDControllerManager.sharedInstance
+  manager.startControllerForResult(source: self, intent: intent, requestCode: RequestText)
+```
+
+Someone, maybe the caller itself must conform the `FRDIntentForResultSendable` protocol to get the feedback result data from the recipient.
+
+```Swift
+  extension ViewController: FRDIntentForResultSendable {
+
+    func onControllerResult(requestCode: Int, resultCode: FRDResultCode, data: Intent) {
+      if (requestCode == RequestText) {
+        if (resultCode == .ok) {
+          let text = data.extra["text"]
+          print("Successful confirm get from destination : \(text)")
+        } else if (resultCode == .canceled) {
+          let text = data.extra["text"]
+          print("Canceled get from destination : \(text)")
+        }
+      }
+    }
+
+  }
+```
+
+The recipient must conform the `FRDIntentForResultReceivable` protocol to receive the Intent. `FRDIntentForResultReceivable` is `FRDIntentReceivable`'s sub-protocol, and has two more variables.
+
+```Swift
+  var data: [String: Any]?
+  var requestCode: Int?
+```
+
+#### Custom transition animation
+
+In FRDIntent, the transition is abstracted as `FRDControllerDisplay` protocol. There are already two implementations: `FRDPushDisplay` and `FRDPresentationDisplay` of `FRDControllerDisplay`.
+
+Your custom transition should conform this protocol `FRDControllerDisplay`. When the caller want to launch a view controller, set the `FRDIntent`'s `controllerDisplay` variable to the object presenting your custom tranistion.
+
+FRDIntent's default transition is `FRDPushDisplay` within `starController`, or `FRDPresentationDisplay` within `startControllerForResult`.
+
+## URLRoutes
+
+FRDIntent/URLRoutes is an URL Router. FRDIntent/URLRoutes sustains a map between url and block. Any external app can send the url request to execute the corresponding block.
```

Comment:
FRDIntent/URLRoutes provides a router from URLs to the app's functional blocks. iOS provides a solution to launch an app from another app based on URL protocols. One may land on different pages based on the path and parameters passed with the URL. FRDIntent/URLRoutes takes advantage of this mechanism, and allows developers to easily define how to invoke functional blocks with URL.